packages/nimble-components/src/select/index.ts

```diff
@@ -665,7 +671,11 @@ export class Select
     public override selectPreviousOption(): void {
         // don't call super.selectPreviousOption as that relies on side-effecty
         // behavior to not select disabled option (which no longer works)
-        for (let i = this.selectedIndex - 1; i >= 0; i--) {
+        for (
+            let i = this.selectedIndex - 1;
+            i >= (this.firstVisibleOptionIndex ?? 0);
+            i--
+        ) {
             if (!this.options[i]?.disabled) {
```

Comment:
the issue is that disabled doesn't mean filtered anymore. Could we instead check for disabled || visually hidden?